File: src/main/java/com/braintreegateway/SubscriptionSearchRequest.java
package com.braintreegateway;

import com.braintreegateway.search.*;

/**
 * Provides a fluent interface to build up requests around {@link Subscription}
 * searches.
 */
public class SubscriptionSearchRequest extends SearchRequest {
    public SubscriptionSearchRequest() {
        super();
    }

    public RangeNode<SubscriptionSearchRequest> daysPastDue() {
        return new RangeNode<SubscriptionSearchRequest>("days_past_due", this);
    }

    public TextNode<SubscriptionSearchRequest> id() {
        return new TextNode<SubscriptionSearchRequest>("id", this);
    }

    public MultipleValueNode<SubscriptionSearchRequest, String> ids() {
        return new MultipleValueNode<SubscriptionSearchRequest, String>("ids", this);
    }

    public MultipleValueNode<SubscriptionSearchRequest, Boolean> inTrialPeriod() {
        return new MultipleValueNode<SubscriptionSearchRequest, Boolean>("in_trial_period", this);
    }

    public MultipleValueNode<SubscriptionSearchRequest, String> merchantAccountId() {
        return new MultipleValueNode<SubscriptionSearchRequest, String>("merchant_account_id", this);
    }

    public DateRangeNode<SubscriptionSearchRequest> nextBillingDate() {
        return new DateRangeNode<SubscriptionSearchRequest>("next_billing_date", this);
    }

    public MultipleValueOrTextNode<SubscriptionSearchRequest, String> planId() {
        return new MultipleValueOrTextNode<SubscriptionSearchRequest, String>("plan_id", this);
    }

    public RangeNode<SubscriptionSearchRequest> price() {
        return new RangeNode<SubscriptionSearchRequest>("price", this);
    }

    public MultipleValueNode<SubscriptionSearchRequest, Subscriptions.Status> status() {
        return new MultipleValueNode<SubscriptionSearchRequest, Subscriptions.Status>("status", this);
    }
    
    public TextNode<SubscriptionSearchRequest> transactionId() {
        return new TextNode<SubscriptionSearchRequest>("transaction-id", this);
    }

    public RangeNode<SubscriptionSearchRequest> billingCyclesRemaining() {
        return new RangeNode<SubscriptionSearchRequest>("billing_cycles_remaining", this);
    }
}

File: src/main/scala/com/braintreegateway/search/SubscriptionSearchRequest.scala
package com.braintreegateway.search

import com.braintreegateway.Subscriptions

/**
 * Provides a fluent interface to build up requests around {@link com.braintreegateway.Subscription}
 * searches.
 */
class SubscriptionSearchRequest extends SearchRequest[SubscriptionSearchRequest] {

  def daysPastDue = rangeNode("days_past_due")

  def id = textNode("id")

  def ids = multipleValueNode[String]("ids")

  def inTrialPeriod = multipleValueNode[Boolean]("in_trial_period")

  def merchantAccountId = multipleValueNode[String]("merchant_account_id")

  def nextBillingDate = dateRangeNode("next_billing_date")

  def planId = multipleValueOrTextNode[String]("plan_id")

  def price = rangeNode("price")

  def status = multipleValueNode[Subscriptions.Status]("status")

  def transactionId = textNode("transaction-id")

  def billingCyclesRemaining = rangeNode("billing_cycles_remaining")

  def getThis = this
}